src/KatesCausalModel/CausalModels.jl

```diff
@@ -0,0 +1,110 @@
+module CausalModels
+
+using Distributions
+using Statistics
+export ExogenousVariable, EndogenousVariable, intervene, randomsample, prob
+
+# Fill in the rest
+struct ExogenousVariable
+    name::Symbol
+    dist::Bernoulli{Float64}
+end
+
+struct EndogenousVariable
+    logical_operator
+    args::Tuple
+end
+
+struct ModifiedEndoVar
+    endo_var::EndogenousVariable
+    dict::Dict
+end
+
+function (v::EndogenousVariable)(u::NamedTuple{(:courtorder, :nervous),Tuple{Int64,Int64}})
```

Comment:
This type signature is too restrictive.  Again, your model is not necessarily going to only have two boolean exogenous values